scripts/clean_osm_data.py

```diff
@@ -338,17 +389,82 @@ def integrate_lines_df(df_all_lines):
         ] = 3
 
-    # one circuit contains 3 cable
+    # one circuit contains 3 cable under a preliminary assumption of an AC line
     df_all_lines.loc[df_all_lines["circuits"].isna(), "circuits"] = (
         df_all_lines.loc[df_all_lines["circuits"].isna(), "cables"] / 3
     )
-    df_all_lines["circuits"] = df_all_lines["circuits"].astype(int)
 
     # where circuits are "0" make "1"
     df_all_lines.loc[
-        (df_all_lines["circuits"] == "0") | (df_all_lines["circuits"] == 0), "circuits"
+        (df_all_lines["circuits"] == "0") | (df_all_lines["circuits"] == 0),
+        "circuits",
     ] = 1
 
+    if df_all_lines["circuits"].dtype != int:
+
+        # it's possible that some df_all_lines["circuits"] are in manually dropped_tags
+        if any(df_all_lines["circuits"].isin(dropped_tags)):
+
+            # reset indexing to avoid 'SettingWithCopyWarning' troubles in further operations with the data frame
+            df_one_third_circuits = df_all_lines.loc[
+                df_all_lines["circuits"].isin(dropped_tags)
+            ].reset_index()
+
+            # avoid mixing column name with geopandas method
+            df_one_third_circuits = df_one_third_circuits.rename(
+                columns={
+                    "length": "length_osm",
+                }
+            )
+
+            # transfrom to EPSG:4326 from EPSG:3857 to obtain length in m from coordinates
+            df_one_third_circuits_m = df_one_third_circuits.set_crs("EPSG:4326").to_crs(
+                "EPSG:3857"
+            )
+
+            length_from_crs = df_one_third_circuits_m.length
+            df_one_third_circuits["length_crs"] = length_from_crs
+
+            # in case a line length is not available directly
+            df_one_third_circuits.loc[
+                df_one_third_circuits["length_osm"].isna(), "length_osm"
+            ] = df_one_third_circuits.loc[
+                df_one_third_circuits["length_osm"].isna(), "length_crs"
+            ]
+
+            # [m] -> [km]
+            dropped_length = round(df_one_third_circuits["length_osm"].sum() / 1e3, 1)
+            dropped_values = set(df_one_third_circuits["circuits"])
+            logger.warning(
+                f"The lines with a circuit tag in {dropped_values} of an overal length {dropped_length} km dropped."
+            )
+
+            # troubles with projections can lead to discrepancy between the length values
+            tol = 0.1  # [m]
+            length_diff = (
+                df_one_third_circuits["length_osm"]
+                - df_one_third_circuits["length_crs"]
+            )
+
+            if any(length_diff > tol):
+                total_length_diff = round(sum(length_diff), 2)
+                logger.warning(
+                    f"There is a difference of {total_length_diff} m in the dropped lines length as compared with values extracted from geographical coordinates."
+                )
```

Comment:
Nice check, though may 0.1m be a too tiny tolerances? That parameter may be an optional argument to this function for generalizability purposes.